src/gurt/shm_dict.c

```diff
@@ -621,13 +619,13 @@ shm_ht_rec_find(d_shm_ht_loc_t shm_ht_loc, const char *key, const int len_key,
 	}
 
 	ht_head    = shm_ht_loc->ht_head;
-	hash       = d_hash_string_u32(key, len_key);
-	idx        = hash & (ht_head->n_bucket - 1);
+	idx        = (unsigned int)(d_hash_murmur64((const unsigned char *)key, len_key, 0) &
```

Comment:
this doesn't make sense to me.. you changes to use a 64 bit hash, but then you cast it back to 32 bit?
what is going on?

Reply:
> The distribution of d_hash_string_u32() is bad and d_hash_murmur64() shows more even distribution in my previous test. If you do not like the change here, we can add murmur32 function and call murmur32() then. Please let me know. Thank you!

i do not understand this comment. what does this have to do with what i like and what i don't like?
this has nothing to do about distribution. yes you should use murmur64, but what does your commend have to do with my comment about not casting it back to 32 bit??

Reply:
@knard38 In the modulous operation "idx = hash % n_bucket", all 64 bit of the hash are used. The result of modulous is smaller than the number of buckets. (uint32_t) is applied on the final result of (hash % n_bucket).